org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/diagnostics/LSPDiagnosticsToMarkers.java

```diff
@@ -167,7 +168,7 @@ private void updateMarkers(PublishDiagnosticsParams diagnostics, IResource resou
 				}
 			}
 		};
-		ResourcesPlugin.getWorkspace().run(runnable, new NullProgressMonitor());
+		ResourcesPlugin.getWorkspace().run(runnable, resource, IWorkspace.AVOID_UPDATE, new NullProgressMonitor());
```

Comment:
The best scheduling rule would be `resource.getWorkspace().getRuleFactory().markerRule(resource)`

Reply:
Ok, I changed it accordingly.